src/main/java/org/traccar/protocol/Gl200TextProtocolDecoder.java

```diff
@@ -1033,36 +1033,34 @@ private Object decodeEri(Channel channel, SocketAddress remoteAddress, String[]
         return positions;
     }
 
-    private static final Pattern PATTERN_IGN = new PatternBuilder()
-            .text("+").expression("(?:RESP|BUFF):GT[IV]G[NF],")
-            .expression("(?:.{6}|.{10})?,")      // protocol version
-            .number("(d{15}|x{14}),")            // imei
-            .expression("[^,]*,")                // device name
-            .number("d+,")                       // ignition off duration
-            .expression(PATTERN_LOCATION.pattern())
-            .number("(d{5}:dd:dd)?,")            // hour meter
-            .number("(d{1,7}.d)?,")              // odometer
-            .number("(dddd)(dd)(dd)")            // date (yyyymmdd)
-            .number("(dd)(dd)(dd)").optional(2)  // time (hhmmss)
-            .text(",")
-            .number("(xxxx)")                    // count number
-            .text("$").optional()
-            .compile();
-
-    private Object decodeIgn(Channel channel, SocketAddress remoteAddress, String sentence) {
-        Parser parser = new Parser(PATTERN_IGN, sentence);
-        Position position = initPosition(parser, channel, remoteAddress);
-        if (position == null) {
+    private Object decodeIgn(
+            Channel channel, SocketAddress remoteAddress, String[] v, String type) throws ParseException {
+        int index = 0;
+        index += 1; // header
+        String protocolVersion = v[index++]; // protocol version
+        DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, v[index++]);
+        if (deviceSession == null) {
             return null;
         }
+        Position position = new Position(getProtocolName());
+        position.setDeviceId(deviceSession.getDeviceId());
 
-        decodeLocation(position, parser);
+        String model = getDeviceModel(deviceSession, protocolVersion);
+        index += 1; // device name
+        index += 1; // duration of ignition on/off
 
-        position.set(Position.KEY_IGNITION, sentence.contains("GN"));
-        position.set(Position.KEY_HOURS, parseHours(parser.next()));
-        position.set(Position.KEY_ODOMETER, parser.nextDouble() * 1000);
+        decodeLocation(position, model, v, index);
```

Comment:
This actually broke decoding:

https://www.traccar.org/forums/topic/gl200-protocol-error-in-gv55w-device/

Fixed it here: https://github.com/traccar/traccar/commit/b1e0484ffd8fe8e71b4027cdb06d23431f5dfac8. Please be more careful with changes in the future.